modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -670,6 +671,26 @@ struct CV_EXPORTS_W HOGDescriptor
     void groupRectangles(std::vector<cv::Rect>& rectList, std::vector<double>& weights, int groupThreshold, double eps) const;
 };
 
+
+class CV_EXPORTS_W QRCodeEncoder {
+public:
+    enum {
+        CORRECT_LEVEL_L = 0,
+        CORRECT_LEVEL_M = 1,
+        CORRECT_LEVEL_Q = 2,
+        CORRECT_LEVEL_H = 3} QRCodeCorrectionLevel;
+    std::vector<Mat> generate(cv::String input_string, int mode, int version, int correction_level,int m, int eci,int s);
+    Mat generate(cv::String input_string, int mode, int version , int correction_level ,int m  , int eci);
+
+    int mode_type;
+    int version_level;
+    int ecc_level;
+    int mask_type;
+    int eci_num;
+    int struct_num;
```

Comment:
There is no need to keep these variables in this class because they use only in class `QREncode`.

Reply:
They are also used in class QRDecode